### PR TITLE
Feature/qt5

### PR DESCRIPTION
**Pull request by hemker**
I added a `rock_find_qt5` macro based on #61 and adapted the macros to work under Ubuntu 20.04 and 18.04. By using the new macro one is able to compile packages using Qt5, as it is used in [rock-replay](https://github.com/rock-cpp/rock_replay/tree/feature/qt5). So far I cannot see any regression while introducing both macros to the users. If it does, I'd be happy to get to know what has to be done to push development further towards Qt5.

### CODE REVIEW

**Review by doudou — APPROVED** [review]

**Comment by doudou** [comment]
Looks good @hemker ... Thanks !

I'll let CI pass before I merge this, but I don't think there should be any regression.

There is a separate build configuration (https://github.com/rock-core/rock.build-tests-buildconf) to implement build-related tests. Would you consider adding test packages ? I could see having two build_tests/cmake/qt4 and build_tests/cmake/qt5 packages which would resp. use `rock_find_qt4` and `rock_find_qt5` and use some Qt4/Qt5 APIs to check that the build is set up right. No need to have something actually working, the goal is only to test the build setup.

**Comment by hemker** [comment]
@doudou Yep, I can implement two of those packages.
Unfortunately, I am not member of the rock-core team, so I cannot add these two repositories in the right name space. If you create an empty repo for both test cases, I can at least provide a PR.

**Comment by hemker** [comment]
@doudou Created respective test packages

**Comment by doudou** [comment]
Omg ... I completely forgot about this. 

I'm having a look at it right now.

File: modules/Rock.cmake
Comment (doudou) [inline modules/Rock.cmake:477]: Indentation looks wrong. Use 4 spaces, no tabs.

File: modules/Rock.cmake
Comment (doudou) [inline modules/Rock.cmake:1240]: The behavior of forcing the addition of some modules in rock_find_qt4 is ... questionable (at best). Maybe we should not replicate it in the new rock_find_qt5 ? What do you think ?

Comment (doudou) [reply modules/Rock.cmake:1240]: Additionally, I would guess that it would be best here to use the CMake imported targets instead of the old-style add_definitions/include_directories/link_directories. `rock_find_qt4` existed from before that time ;-)

**Comment by doudou** [comment]
Hi @hemker

Again, I'm very sorry for having forgotten that pull request ... The CI part is alright. However, I spotted some stuff re-reviewing it.

I'm going for vacations for two weeks. Will have a look again afterwards.

File: modules/Rock.cmake
Comment (hemker) [reply modules/Rock.cmake]: Resolved.

File: modules/Rock.cmake
Comment (hemker) [reply modules/Rock.cmake:1240]: Yes, this is a good point. I think we shouldn't enforce any kind of library dependency here. It is also possible to only use e.g. Qt5::Widgets, so I'm going to remove the hardcoded deps.

Comment (hemker) [reply modules/Rock.cmake]: I completely removed the old-fashioned include and link directories. They are not necessary as dependencies are set up correctly via the DEPS argument in rock_executable and rock_library.

**Comment by hemker** [comment]
Hi @doudou 
I proposed solutions to all open points. Have a nice holiday and I hope we can get this done after it :)

File: modules/Rock.cmake
Comment (doudou) [inline modules/Rock.cmake:487]: Debug output ...

File: modules/Rock.cmake
Comment (hemker) [reply modules/Rock.cmake]: @doudou Converted all tabs to spaces throughout the whole file. Could also offer to run cmake-format beforehand to bring some better indentation to the Rock.cmake file.

File: modules/Rock.cmake
Comment (doudou) [inline modules/Rock.cmake:467]: You converted tabs to spaces, but did so while your editor was configured to use 4 spaces per tab. Indentation is still broken. Please double-check on the GH pull request interface (which uses 8-spaces tabs)

Comment (doudou) [reply modules/Rock.cmake:467]: I only highlighted this, but there are other places with the same problem.

Comment (hemker) [reply modules/Rock.cmake]: @doudou Sorry for that, have been to inattentive with it. Now I cannot see any wrongly indented lines in the Github diff for the PR.

**Comment by hemker** [comment]
@doudou Is there anything to do yet?

**Comment by doudou** [comment]
Nope. Thanks for your patience @hemker

**Pull request merged**
